worlds/ahit/Rules.py

```diff
@@ -0,0 +1,988 @@
+
+        for misc in data.misc_required:
+            add_rule(location, lambda state, item=misc: state.has(item, world.player))
+
+    set_specific_rules(world)
+
+    # Putting all of this here, so it doesn't get overridden by anything
+    # Illness starts the player past the intro
+    alpine_entrance = world.multiworld.get_entrance("AFR -> Alpine Skyline Area", world.player)
+    add_rule(alpine_entrance, lambda state: can_use_hookshot(state, world))
+    if world.options.UmbrellaLogic.value > 0:
+        add_rule(alpine_entrance, lambda state: state.has("Umbrella", world.player))
+
+    if zipline_logic(world):
+        add_rule(world.multiworld.get_entrance("-> The Birdhouse", world.player),
+                 lambda state: state.has("Zipline Unlock - The Birdhouse Path", world.player))
+
+        add_rule(world.multiworld.get_entrance("-> The Lava Cake", world.player),
+                 lambda state: state.has("Zipline Unlock - The Lava Cake Path", world.player))
+
+        add_rule(world.multiworld.get_entrance("-> The Windmill", world.player),
+                 lambda state: state.has("Zipline Unlock - The Windmill Path", world.player))
+
+        add_rule(world.multiworld.get_entrance("-> The Twilight Bell", world.player),
+                 lambda state: state.has("Zipline Unlock - The Twilight Bell Path", world.player))
+
+        add_rule(world.multiworld.get_location("Act Completion (The Illness has Spread)", world.player),
+                 lambda state: state.has("Zipline Unlock - The Birdhouse Path", world.player)
+                 and state.has("Zipline Unlock - The Lava Cake Path", world.player)
+                 and state.has("Zipline Unlock - The Windmill Path", world.player))
+
+    if zipline_logic(world):
+        for (loc, zipline) in zipline_unlocks.items():
+            add_rule(world.multiworld.get_location(loc, world.player),
+                     lambda state, z=zipline: state.has(z, world.player))
+
+    for loc in world.multiworld.get_region("Alpine Skyline Area (TIHS)", world.player).locations:
+        if "Goat Village" in loc.name:
+            continue
+        # This needs some special handling
+        if loc.name == "Alpine Skyline - Goat Refinery":
+            add_rule(loc, lambda state: state.has("AFR Access", world.player)
+                     and can_use_hookshot(state, world)
+                     and can_hit(state, world, True))
+
+            difficulty: Difficulty = Difficulty(world.options.LogicDifficulty.value)
+            if difficulty >= Difficulty.MODERATE:
+                add_rule(loc, lambda state: state.has("TIHS Access", world.player)
+                         and can_use_hat(state, world, HatType.SPRINT), "or")
+            elif difficulty >= Difficulty.HARD:
+                add_rule(loc, lambda state: state.has("TIHS Access", world.player, "or"))
+
+            continue
+
+        add_rule(loc, lambda state: can_use_hookshot(state, world))
```

Comment:
I get Goat Refinery being special, but the rest of the non-Village TIHS locations seem like they could just have `hookshot=True` over in Locations.py.

Reply:
Done